framework/auth/views.py

```diff
@@ -133,71 +112,92 @@ def reset_password_post(auth, verification_key=None, **kwargs):
-        forms.push_errors_to_status(form.errors)
-        # Don't go anywhere
 
     return {
-        'verification_key': verification_key
-    }, 400
+        'uid': user_obj._id,
+        'token': user_obj.verification_key_v2['token'],
+    }
 
 
 @collect_auth
-def forgot_password_post(auth, **kwargs):
+def forgot_password_get(auth):
     """
-    View for user to submit forgot password form.
-    HTTP Method: POST
+    View for user to land on the forgot password page.
+    HTTP Method: GET
+
+    :param auth: the authentication context
+    :return
     """
 
-    # If user is already logged in, redirect to dashboard page.
+    # if users are logged in, log them out and redirect back to this page
     if auth.logged_in:
-        return redirect(web_url_for('dashboard'))
+        return auth_logout(redirect_url=request.url)
+
+    return {}
+
+
+def forgot_password_post():
+    """
+    View for user to submit forgot password form.
+    HTTP Method: POST
+    :return {}
+    """
 
     form = ForgotPasswordForm(request.form, prefix='forgot_password')
 
-    if form.validate():
+    if not form.validate():
+        # Don't go anywhere
+        forms.push_errors_to_status(form.errors)
+    else:
         email = form.email.data
         status_message = ('If there is an OSF account associated with {0}, an email with instructions on how to '
                           'reset the OSF password has been sent to {0}. If you do not receive an email and believe '
                           'you should have, please contact OSF Support. ').format(email)
         # check if the user exists
         user_obj = get_user(email=email)
-        if user_obj:
-            # check forgot_password rate limit
-            if throttle_period_expired(user_obj.email_last_sent, settings.SEND_EMAIL_THROTTLE):
-                # new random verification key, allows OSF to check whether the reset_password request is valid,
-                # this verification key is used twice, one for GET reset_password and one for POST reset_password
-                # and it will be destroyed when POST reset_password succeeds
-                user_obj.verification_key = generate_verification_key()
-                user_obj.email_last_sent = datetime.datetime.utcnow()
-                user_obj.save()
-                reset_link = furl.urljoin(
-                    settings.DOMAIN,
-                    web_url_for(
-                        'reset_password_get',
-                        verification_key=user_obj.verification_key
-                    )
-                )
-                mails.send_mail(
-                    to_addr=email,
-                    mail=mails.FORGOT_PASSWORD,
-                    reset_link=reset_link
-                )
-                status.push_status_message(status_message, kind='success', trust=False)
-            else:
+        if not user_obj:
+            # do not reveal user existence information by pushing success message even when user not found
```

Comment:
This can be accomplished without code duplication by putting the call to `push_status_message` outside this conditional